### ISSUE TITLE
Feature Request: Bare repositories management

### ISSUE THREAD

**Issue by MathiasSM** [ForkIssues/Tracker#247]
I would like to be able to manage bare repositories the same way I manage regular repositories in this app. I use bare repositories to keep track of scattered files, usually configuration files. There are other personal uses (besides git servers, in which all repos are bare repos)

I think the main things to consider here to implement the feature is 1. the ability to ignore untracked files, 2. adding a repo not by working dir but by .git (which in this case can by .anything), and 3. just check current changes to any file listed in the git db? Dunno, I know very little about git internals. Would love to take a look at how this works, though, to understand why only so few git GUIs implement it.

**Comment by brechtm** [comment]
If you just want to be able to browse the history for the bare repository, you can create a clone without creating a checkout:
```
git clone --no-checkout /path/to/repo.git repo
```
You'll still need to fetch to see changes to the bare repository though.

**Comment by brechtm** [comment]
A first test with [forkgit](https://github.com/brechtm/forkgit) seems somewhat succesful:

1. clone the forkgit repository and configure Fork to use its `bin/git` as a custom git instance
2. `git clone --bare <some-git-repository>`
3. inside the bare repository directory create a `.forkgit` file with these contents:
   ```git-dir = .```
4. create an empty `.git` directory in the bare repository directory
5. open the cloned repository in Fork

This allows browsing the commit log. _Local Changes_ just shows no files.

Updating the bare repository from a remote requires extra steps. See [How do I update my bare repo?](https://stackoverflow.com/questions/3382679/how-do-i-update-my-bare-repo) on StackOverflow. After fetching Fork doesn't register that the local branch has moved.

Note that forkgit has only been tested on macOS.

**Comment by bartvanandel** [comment]
I'm not quite sure why you'd need a work tree for this to work, at least for showing history. Seeing as `git log` works fine, and `gitk` (ages old git GUI) has no problem showing git history including diffs for bare repos, what's really keeping Fork from adding basic support for bare repos?

**Comment by DanPristupov** [comment]
> what's really keeping Fork from adding basic support for bare repos?

Our time. I want the Fork source code to be simpler and maintenance to be cheaper.

**Comment by bartvanandel** [comment]
Fair point, but how much different is showing info from a bare repo really? Aren't the commands (or lib calls) just the same?

**Comment by DanPristupov** [comment]
> how much different is showing info from a bare repo really?

Working directory becomes optional.

> Aren't the commands (or lib calls) just the same?

2/3 of features must be updated to handle that case. The commit view must be blocked. Pull must be blocked. Tab headers can't show uncommitted changes. Custom commands must be reworked. Submodules must be reworked. Even the create branch dialog must be updated and not allow checkout.

...

While over the past 8 years, only 4 people have shown interest in this feature.

**Comment by bartvanandel** [comment]
Thanks for the insight!

> only 4 people have shown interest in this feature.

There are 8 thumbs up on the initial post, including yours. And I'm sure there are more people who just didn't bother trying to get attention about this. But sure, not many in total.

**Comment by DanPristupov** [comment]
Yes, you are right.

Still, right now, there are more important tasks: performance, git3.0 (SHA-256, reftables), interactive rebase improvements, etc.
